### ISSUE TITLE
Model Accuracy

### ISSUE THREAD

**Issue by Wayn-Git** [Wayn-Git/CatvsDog#2]
Model currently hits 80% test accuracy.
    Not bad, but it can definitely be improved, especially since the dataset is solid (4k cats + 4k dogs, 2k test).

✅ Updates

    Looked into pretrained models (MobileNetV2, EfficientNet, etc.) — they’re a better option for performance and training time.

    Found that using transfer learning with include_top=False and freezing the base model gives faster and better results.

❌ Issue (Fixed)

Ran into this error:

Input 0 of layer "dense" is incompatible with the layer: expected axis -1 of input shape to have value 50176, but received input with shape (1, 20736)

Stupid mistake — I forgot to flatten the output before feeding it into the dense layer.
Fixed by adding a Flatten() or GlobalAveragePooling2D() after the conv layers.
🔧 Next Steps

    Try out MobileNetV2 with input_shape=(160,160,3) and a custom classifier head.

    Use GlobalAveragePooling2D() to avoid shape mismatch issues.

    Add EarlyStopping and ImageDataGenerator or tf.data for efficient training.

    Target: 90–95% accuracy without overfitting.

**Comment by Wayn-Git** [comment]
## ✅ **SOLVED** - Transfer Learning + Smart Callbacks

### 🔧 **Issue & Fix**
Shape mismatch error when connecting conv layers to dense layers. **Fixed by using `GlobalAveragePooling2D()`** instead of Flatten().

```python
# Solution: MobileNetV2 + GlobalAveragePooling2D
base_model = MobileNetV2(input_shape=(160, 160, 3), include_top=False, weights='imagenet')
base_model.trainable = False

model = Sequential([
    base_model,
    GlobalAveragePooling2D(),  # ← KEY FIX: Eliminates shape issues
    Dropout(0.3),
    Dense(128, activation='relu'),
    Dropout(0.5),
    Dense(1, activation='sigmoid')
])
```

### 🚀 **Results**
- **Accuracy**: 80% → **95%+** 
- **Training**: 30 epochs → **15-20 epochs**
- **Stability**: Much better convergence

### 🎯 **Smart Callbacks Used**
```python
callbacks = [
    EarlyStopping(monitor='val_accuracy', patience=5, restore_best_weights=True),
    ReduceLROnPlateau(monitor='val_loss', factor=0.2, patience=3)
]
```

**Key insight**: Transfer learning + proper layer transitions + intelligent callbacks = faster, better results 🚀
